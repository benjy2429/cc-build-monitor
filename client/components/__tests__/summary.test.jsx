import React from 'react';
import { shallow, mount } from 'enzyme';
import Summary from '../summary';

const defaultProps = {
  projects: [
    { name: 'org/project-1' },
    { name: 'org/project-2' },
  ],
  category: 'passing',
};

describe('Summary', () => {
  it('renders correctly', () => {
    const component = shallow(
      <Summary {...defaultProps} />,
    );

    expect(component).toMatchSnapshot();
  });

  it('does not render with no projects', () => {
    const component = shallow(
      <Summary projects={[]} />,
    );

    expect(component.html()).toEqual(null);
  });

  it('displays the second project after a timeout', () => {
    jest.useFakeTimers();
    const component = mount(
      <Summary {...defaultProps} />,
    );
    jest.runOnlyPendingTimers();
    expect(component).toMatchSnapshot();
  });

  it('loops back to the first project', () => {
    jest.useFakeTimers();
    const component = mount(
      <Summary {...defaultProps} />,
    );
    jest.runOnlyPendingTimers();
    jest.runOnlyPendingTimers();
    expect(component).toMatchSnapshot();
  });

  it('strips the org when specified', () => {
    config.stripOrgs = true;
    const component = shallow(
      <Summary {...defaultProps} />,
    );

    expect(component).toMatchSnapshot();
  });
});

