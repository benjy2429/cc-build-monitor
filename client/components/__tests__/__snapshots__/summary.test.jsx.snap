// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary displays the second project after a timeout 1`] = `
<Summary
  category="passing"
  projects={
    Array [
      Object {
        "name": "org/project-1",
      },
      Object {
        "name": "org/project-2",
      },
    ]
  }
>
  <div
    className="summary summary-passing"
  >
    <div
      className="summary-heading"
    >
      2
       
      passing
    </div>
    <div
      className="summary-details"
    >
      org/project-2
    </div>
  </div>
</Summary>
`;

exports[`Summary loops back to the first project 1`] = `
<Summary
  category="passing"
  projects={
    Array [
      Object {
        "name": "org/project-1",
      },
      Object {
        "name": "org/project-2",
      },
    ]
  }
>
  <div
    className="summary summary-passing"
  >
    <div
      className="summary-heading"
    >
      2
       
      passing
    </div>
    <div
      className="summary-details"
    >
      org/project-1
    </div>
  </div>
</Summary>
`;

exports[`Summary renders correctly 1`] = `
<div
  className="summary summary-passing"
>
  <div
    className="summary-heading"
  >
    2
     
    passing
  </div>
  <div
    className="summary-details"
  >
    org/project-1
  </div>
</div>
`;
